src/main/java/edu/kis/powp/jobs2d/command/DriverCommandVisitor.java

```diff
@@ -0,0 +1,31 @@
+package edu.kis.powp.jobs2d.command;
+
+import java.util.Iterator;
+
+
+/**
+ * Visitor class for driver commands.
+ */
+public class DriverCommandVisitor implements Visitor {
+    private int counter = 0;
+
+    public int getCounter(){
+        return this.counter;
+    }
+
+    @Override
+    public void visit(DriverCommand driverCommand) {
+        counter++;
+        System.out.println("Counter: " + counter);
+    }
+
+    @Override
+    public void visit(ICompoundCommand compoundCommand) {
+        Iterator<DriverCommand> iterator = compoundCommand.iterator();

```

Comment:
This solution will not be correct if compound command contains another compound command. 
There is a generic way to solve it using visitor in the context of composite.